oak-store-spi/src/main/java/org/apache/jackrabbit/oak/plugins/memory/ModifiedNodeState.java

```diff
@@ -163,9 +164,9 @@ static Iterable<? extends PropertyState> getProperties(
             final Set<String> keys = properties.keySet();
             Predicate<PropertyState> predicate =
                     x -> !keys.contains(x == null ? null : x.getName());
-            return concat(
+            return IterableUtils.chainedIterable(
                     filter(base.getProperties(), predicate::test),
-                    filter(properties.values(), x -> x != null));
+                    filter(properties.values(), Objects::nonNull));
```

Comment:
This change seems unrelated.

Reply:
addressed in https://github.com/apache/jackrabbit-oak/pull/2049/commits/71020cc5ef9c154589a554c5ab83d68dd4187b80